### PR TITLE
Remove examples and move example feature to Game of Life

### PR DESCRIPTION
**Pull request by pbirkhol-ni**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

#534 

- Remove the `ui_progress_updates` example since UI progress updates are now demonstrated in the `game_of_life` example.
- Remove the `sample_streaming_measurement` example. This was originally meant to be used for tests, but that never ended up happening. Now it is just a maintenance burden that doesn't add any value over the other examples.
- Add cancellation support to the `game_of_life` example. Also, fix the example to wait for Update Interval - Actual Update Time. This makes the update interval more accurate since it can take > 50msec to update the board.

### What testing has been done?

Ran the `game_of_life` example on my machine.

### CODE REVIEW

File: examples/game_of_life/_helpers.py
Comment (pbirkhol-ni) [inline examples/game_of_life/_helpers.py:1]: The changes in this file were from black/ni-style-guide. I don't make any real changes. I'm not sure why black decided to reformat this, though.

**Comment by bkeryan** [comment]
> Remove the sample_streaming_measurement example. This was originally meant to be used for tests, but that never ended up happening.

It did happen, but then we forked it so the tests are self-contained.

File: examples/game_of_life/_helpers.py
Comment (bkeryan) [reply examples/game_of_life/_helpers.py:1]: Did you run black directly or through ni-python-styleguide? The example projects don't yet configure [tool.black] line-length = 100, so running black directly produces different results than running it through ni-python-styleguide. 

The weird part is that ni-python-styleguide lint doesn't fail with either format. Perhaps flake8-black & black --check are tolerant of some formatting differences (at least when adding/removing a trailing comma).

Cc: @mshafer-NI

**Review by bkeryan — CHANGES_REQUESTED** [review]

**Review by dixonjoel — APPROVED** [review]

File: examples/game_of_life/_helpers.py
Comment (pbirkhol-ni) [reply examples/game_of_life/_helpers.py:1]: I did run black directly first. Then I ran ni-style-guide fix. I will revert this file and only run ni-style-guide.

File: examples/game_of_life/measurement.py [hunk 1]
```diff
@@ -32,9 +34,13 @@ def measure(
     width: int, height: int, update_interval: int, max_generations: int
 ) -> Generator[Tuple[xydata_pb2.DoubleXYData, int], None, None]:
     """Streaming measurement that returns Conway's Game of Life grid as DoubleXYData."""
+    cancellation_event = threading.Event()
+    measurement_service.context.add_cancel_callback(cancellation_event.set)
     grid = _initialize_grid_with_seeded_data(width, height)
+    update_interval_sec = update_interval / 1000
```
[the remark this answers is not in the document]
Comment (pbirkhol-ni) [reply examples/game_of_life/measurement.py:40]: It's easy enough to change so I made the variable `update_interval_msec` and then rebinded the measui control.

**Review by bkeryan — APPROVED** [review]

**Pull request merged**